projects/pastanaga-angular/src/lib/side-nav/side-nav.component.spec.ts

```diff
@@ -0,0 +1,121 @@
+import { CommonModule } from '@angular/common';
+import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
+import { MockModule } from 'ng-mocks';
+import { SideNavComponent } from './side-nav.component';
+import { PaButtonModule } from '../button/button.module';
+import { ReplaySubject, Subject } from 'rxjs';
+import { TranslatePipe } from '../translate/translate.pipe';
+import { PastanagaService } from '../pastanaga.service';
+import { createSpyObject, mockProvider } from '@ngneat/spectator';
+import { BreakpointObserver, ViewportMode } from '../breakpoint-observer/breakpoint.observer';
+import { fakeAsync, tick } from '@angular/core/testing';
+
+describe('SideNavComponent', () => {
+    let spectator: Spectator<SideNavComponent>;
+    let component: SideNavComponent;
+    const currentMode: ReplaySubject<ViewportMode> = new ReplaySubject<ViewportMode>(1);
+    const toggle: Subject<boolean> = new Subject<boolean>();
+    const createComponent = createComponentFactory({
+        component: SideNavComponent,
+        imports: [MockModule(CommonModule), MockModule(PaButtonModule)],
+        mocks: [TranslatePipe],
+        providers: [
+            mockProvider(PastanagaService, {
+                breakpoint: createSpyObject(BreakpointObserver, {
+                    currentMode,
+                }),
```

Comment:
pastanagaService is like oneService: that was a false good idea which should be avoided. Replace the use of PastanagaService by what you're using inside it (BreakpointObserver in your case)